lightning/src/util/events.rs

```diff
@@ -67,6 +68,38 @@ pub enum PaymentPurpose {
 	SpontaneousPayment(PaymentPreimage),
 }
 
+#[derive(Clone, Debug)]
+/// Some information provided on the closure source of the channel halting.
+pub enum ClosureReason {
+	/// Closure generated from receiving a peer error message by ChannelManager::handle_error
+	CounterpartyForceClosed {
+		/// The error is coming from the peer, there *might* be a human-readable msg
+		peer_msg: Option<String>,
+	},
+	/// Closure generated from ChannelManager::force_close_channel
+	HolderForceClosed,
+	/// Closure generated from receiving a peer's ClosingSigned message. Note the shutdown
+	/// sequence might have been initially initiated by us.
+	CooperativeClosure,
+	/// Closure generated from receiving chain::Watch's CommitmentTxBroadcast event.
```

Comment:
This can also happen in ChannelManager. Probably good to focus on the "what" not the "where in the code". Maybe "A commitment transaction was confirmed on chain, closing the channel. Most likely this commitment transaction came from our counterparty, but it may also have come from a copy of our own `ChannelMonitor`."